### PR TITLE
Merge frontend-rest to Master

### PR DESCRIPTION
**Pull request by dillonclaremont**
- Converts all UI pages to leverage REST
- added styling to all pages
- added RecommendationService to UI

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -15,7 +15,7 @@
   </component>
   <component name="JavacSettings">
     <option name="ADDITIONAL_OPTIONS_OVERRIDE">
-      <module name="BookStore" options="" />
```
Comment (kajhemmingsen) [inline .idea/compiler.xml:18]: Same as when I tried to merge, is this necessary?

File: src/main/java/controllers/BookstoreController.java [hunk 2]
```diff
@@ -93,18 +126,11 @@ public Bookstore createBookstore(@RequestBody Bookstore bookstore){
         return b.getSales();
     }
 
-
-    /**
```
Comment (kajhemmingsen) [inline src/main/java/controllers/BookstoreController.java:97]: Comments got deleted here again?

File: src/main/resources/static/js/shopBookstore.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                    alert("Book is already in cart");
+                }
+            });
+        
+        }
+
+        populateBooks();
+        populateBookstoreInfo();
+        populateCustomerInfo();
+    })
```
Comment (kajhemmingsen) [inline src/main/resources/static/js/shopBookstore.js:135]: Javascript doesn't like not having a newline at the end of .js files, don't think theres any functional penalties though

**Comment by dillonclaremont** [comment]
> This may be the greatest web app the world has ever seen. Great work everyone.

amazon.... here we come

File: src/main/resources/templates/index.html [hunk 4]
```diff
@@ -4,66 +4,76 @@
 <head>
     <title>BookStore</title>
     <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
-    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
+    <script src='https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js'></script>
+    <script src="js/scripts.js"></script>
+    <link rel="stylesheet" href="css/editBookstore.css">
+    <script>
```
Comment (kajhemmingsen) [inline src/main/resources/templates/index.html:10]: scripts should be in scripts.js no?

**Pull request merged**